hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1237,15 +1239,43 @@ public AbfsInputStream openFileForRead(final Path path,
 
       String relativePath = getRelativePath(path);
 
-      final AbfsRestOperation op = client
-          .getPathStatus(relativePath, false, tracingContext);
+      AbfsRestOperation op = null;
+      if (getPrefixMode() == PrefixMode.BLOB) {
+        try {
+          op = client.getBlobProperty(new Path(relativePath), tracingContext);
+        } catch (AbfsRestOperationException e) {
+          if (e.getStatusCode() != HTTP_NOT_FOUND) {
+            throw e;
+          }
+          List<BlobProperty> blobsList = getListBlobs(new Path(relativePath), null,
+                  tracingContext, 2, true);
+          if (blobsList.size() > 0) {
```

Comment:
we should throw 404 if bloblist.size==0

Reply:
on a different note, do we even need to check bloblist when the required blob is not there. Its making no good. Only leading to 404. Why not leave blobListAPI call as it has nothing good to do here.

Reply:
This is the case for implicit dir checking. GetBlobProperty would fail for an existing implicit Dir, but it does not mean that the blob is actually not present.

Reply:
cc for discussion: @snvijaya @anmolanmol1234 @anujmodi2021

Reply:
Keeping the GetListBlobs call for now, for need of different error message in case of implicit directory and an actually non-existent resource.